mrcnn/model.py

```diff
@@ -3114,8 +3144,12 @@ def unmold_outputs(detections, mrcnn_class, mrcnn_bbox, mrcnn_mask,
             "scores": final_scores,
             "masks": final_masks,
         })
+
+        if image_class:
+            results["image_class"] = image_class[i]
```

Comment:
This does not work. The function operates on batches, so `results` is a list of images. You must append to the dict in the loop:
```suggestion
            "image_class": image_class[i]
        })
```